tm2/pkg/bft/privval/signer/remote/server/options.go

```diff
@@ -0,0 +1,134 @@
+package server
+
+import (
+	"fmt"
+	"log/slog"
+	"net"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/gnolang/gno/tm2/pkg/bft/types"
+	"github.com/gnolang/gno/tm2/pkg/crypto/ed25519"
+	osm "github.com/gnolang/gno/tm2/pkg/os"
+)
+
+// RemoteSignerServer provides a service that forwards requests to a types.Signer.
+type RemoteSignerServer struct {
+	// Required config.
+	signer          types.Signer
+	listenAddresses []string
+	logger          *slog.Logger
+
+	// Optional connection config.
+	keepAlivePeriod time.Duration // If 0, keep alive is disabled.
+	responseTimeout time.Duration // If 0, no timeout is set. Requests reception is not timed out.
+
+	// Optional authentication config.
+	serverPrivKey  ed25519.PrivKeyEd25519  // Default is a random key.
+	authorizedKeys []ed25519.PubKeyEd25519 // If empty, all keys are authorized.
+
+	// Internal.
+	listeners     []net.Listener
```

Comment:
I think there is maybe one oversight with the server implementation, which is the ability to support multiple clients.

In theory, the remote signer server serves only 1 node. It should, in any case, because the keys are not shared between different instances of the validator (there is only 1).

This constraint simplifies the implementation significantly

Reply:
I wasn't aware of that, and it even seemed to me that it was the main reason to revert from the gnokms -> validator connection model, but it wouldn't have made any difference to keep it as it is if all the connections are meant to be 1to1.

Is there no scenario where validators validating different chains would want to use the same identity key? Do you want me to refactor the code to remove that?

Reply:
> Is there no scenario where validators validating different chains would want to use the same identity key?

They can reuse the same identity key, but they should run separate instances of the kms for separate chains, to preserve segmentation

Reply:
Done https://github.com/gnolang/gno/pull/3554/commits/8c031d720b2e5007bc0778742c09cffe543cb4ce